ARExperiment/ARExperiment/SizeComparisonViewController.swift

```diff
@@ -58,21 +102,27 @@ class SizeComparisonViewController: UIViewController, ARExperimentSessionHandler
     }
     
     private func removeNodeIfExistAlready() {
-        sceneView.scene.rootNode.childNode(withName: currentCubeName, recursively: true)?.removeFromParentNode()
-        sceneView.scene.rootNode.childNode(withName: currentTextName, recursively: true)?.removeFromParentNode()
+
+        sceneView.scene.rootNode.childNode(withName: currentCube.filename, recursively: true)?.removeFromParentNode()
+        sceneView.scene.rootNode.childNode(withName: currentCube.textFilename, recursively: true)?.removeFromParentNode()
     }
     
     @IBAction func segmentHasBeenChanged(_ sender: UISegmentedControl) {
+        
+        let segmentedControlToDeselect: UISegmentedControl! =
+            (sender == self.ARMeasuresSegmentedControl) ? self.realWorldMeasuresSegmentedControl : self.ARMeasuresSegmentedControl
+        
+        segmentedControlToDeselect.selectedSegmentIndex = UISegmentedControlNoSegment
+        
+        self.currentScale = (sender == self.ARMeasuresSegmentedControl) ? .arWorldScale : .realWorldScale
```

Comment:
Puedes pasar esto a una funcion o variable? no es muy legible la logica que esta haciendo y porque.

Reply:
Yeah the code was not so clear and also having two segmented control that shows the same options its confusing for the user. So I have replaced the two segmented controls with a select world and a select size segmented controls